### PR TITLE
Fix bug virtio port bus is off

### PR DESCRIPTION
**Pull request by xutian**
If bus is False ask qemu to decide bus of the virtio port device to avoid qdevice convert false to off when call set_param;

ID:1403664

### CODE REVIEW

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -491,19 +491,17 @@ def add_virtio_port(devices, chardev, params, name, bus, index=None):
...
                             key, value = _, "NO_EQUAL_STRING"
                         else:
                             key, value = _.split('=')
-                        options[key] = value
+                        virtio_port.set_param(key, value)
                     except Exception:
-                        options.clear()
                         msg = ("Invaild params %s in " % _ +
                                "'virtio_port_param' = %s" % extra_params)
                         logging.error(msg)
```
Comment (suqinhuang) [inline virttest/qemu_vm.py]: need to update "return options" on line 510

Comment (xutian) [inline virttest/qemu_vm.py]: thanks your comments and commit updated, would like to review again.

thanks,
Xu

**Comment by MiriamDeng** [comment]
Hi @xutian 
    I run it on my ppc host but it seems there is some failures
Reproduced traceback from: /root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py:420
Traceback (most recent call last):
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 196, in runTest
    self._runTest()
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 327, in _runTest
    params = env_process.preprocess(self, params, env)
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
    return fn(*args, **kwargs)
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/env_process.py", line 851, in preprocess
    process(test, params, env, preprocess_image, preprocess_vm)
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/env_process.py", line 546, in process
    _call_vm_func()
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/env_process.py", line 515, in _call_vm_func
    vm_func(test, vm_params, env, vm_name)
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/env_process.py", line 183, in preprocess_vm
    migration_exec_cmd=params.get("migration_exec_cmd_dst"))
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
    return fn(*args, **kwargs)
  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/virttest/qemu_vm.py", line 2670, in create
    raise virt_vm.VMStartError(self.name, 'Error occurred while '
VMStartError: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_create_command(). Check the log for traceback.

Traceback (most recent call last):

  File "/root/test_home/mdeng/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 434, in _run_avocado
    raise test_exception

VMStartError: VM 'avocado-vt-vm1' failed to start: Error occurred while executing make_create_command(). Check the log for traceback.

**Comment by suqinhuang** [comment]
@MiriamDeng I guess you didn't use latest code. it works to me

**Comment by suqinhuang** [comment]
ack

**Comment by MiriamDeng** [comment]
@suqinhuang 
 Yes,after my comments xutian adjust the code.

**Comment by MiriamDeng** [comment]
ACK

**Comment by xutian** [comment]
thanks @MiriamDeng  and @suqinhuang

**Pull request merged**